engine/schema/src/main/resources/META-INF/db/schema-41610to41700.sql

```diff
@@ -220,6 +220,19 @@ CREATE VIEW `cloud`.`service_offering_view` AS
     GROUP BY
         `service_offering`.`id`;
 
+
+DELIMITER //
+CREATE PROCEDURE create_external_uuid_in_volumes()
+  BEGIN
+    SELECT count(*) INTO @count FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name = 'volumes' AND table_schema='cloud' AND column_name = 'external_uuid';
+    IF @count < 1 THEN ALTER TABLE cloud.volumes ADD COLUMN external_uuid VARCHAR(40) DEFAULT null;
+    END IF;
+  END //
+DELIMITER ;
+CALL create_external_uuid_in_volumes;
+DROP PROCEDURE create_external_uuid_in_volumes;
```

Comment:
How about this?

````
DROP PROCEDURE IF EXISTS `cloud`.`create_external_uuid_in_volumes`();

CREATE PROCEDURE `cloud`.`create_external_uuid_in_volumes`()
BEGIN
        IF NOT EXISTS ((SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE
            table_name = 'volumes' AND
            table_schema='cloud' AND
            column_name = 'external_uuid'))
        THEN
            ALTER TABLE cloud.volumes ADD COLUMN external_uuid VARCHAR(40) DEFAULT null;
            END IF; END;;

CALL `cloud`.`create_external_uuid_in_volumes`();
````

Reply:
Nope, get also a syntax error. If it work by you, I would suggest that we add this, so that we have progress here. I will check it later what the problem with my env is.

Reply:
@DK101010 
since 4.17 is not released yet, you just need to add line
```
ALTER TABLE `cloud`.`volumes` ADD COLUMN `external_uuid` VARCHAR(40) DEFAULT NULL COMMENT 'UUID of the xxxxxx';
```

Reply:
@weizhouapache ehhhh what ? this was my first solution, What is with idem potent?

Reply:
@DK101010 
if you use stock cloudstack, the table `volumes` does not have the column,  there is no issue.

if you have already added the column in your environment, you can use mysql procedure IDEMPOTENT_ADD_COLUMN which is defined in schema-41520to41600.sql.

please refer to 

https://github.com/apache/cloudstack/blob/e339c4b964257c22bde238708f056bd34621446b/engine/schema/src/main/resources/META-INF/db/schema-41520to41600.sql#L127-L143

Reply:
@DK101010 @weizhouapache thanks I was trying to do the same but forgot the procedure was already set. I would be +1 on using this procedure, in most cases adding the column won't be an issue, but using the idempotent addition we can prevent manual DB intervention on failing upgrades (if somehow an upgrade fails then if you execute the upgrade again all the columns that were added will fail the upgrade since cannot be added again, this can be prevented using the idempotent add)

Reply:
Hi @nvazquez @weizhouapache, I will add this.